Multisig/Data/Services/Safe Client Gateway Service/SCG.swift

```diff
@@ -12,6 +12,49 @@ import Foundation
 enum SCG {}
 
 extension SCG {
+    struct TransactionSummaryItemDateLabel: Decodable {
+        let timestamp: Int
+    }
+
+    struct TransactionSummaryItemLabel: Decodable {
+        let label: String
+    }
+
+    struct TransactionSummaryItemTransaction: Decodable {
+        let transaction: TxSummary
+        let conflictType: ConflictType
+    }
+
+    struct TransactionSummaryItemConflictHeader: Decodable {
+        let nonce: UInt256String
+    }
+
+    enum TransactionSummaryItem: Decodable {
+        case dateLabel(TransactionSummaryItemDateLabel)
+        case label(TransactionSummaryItemLabel)
+        case transaction(TransactionSummaryItemTransaction)
+        case conflictHeader(TransactionSummaryItemConflictHeader)
+        case unknown
+
+        init(from decoder: Decoder) throws {
+            enum Keys: String, CodingKey { case type }
+            let container = try decoder.container(keyedBy: Keys.self)
+            let type = try container.decode(TransactionSummaryItemType.self, forKey: .type)
```

Comment:
I would use plan strings in the Switch instead of the enum because if new enum strings are introduced on the backend, then the app would just decode them as unknown (in default case), otherwise this line will throw exception

Reply:
In that case how about make type as optional and add default case to switch ?

Reply:
works for me but then it is inconsistent with other code